backend/routes/quests.go

```diff
@@ -211,6 +220,103 @@ func GetMainUserQuests(w http.ResponseWriter, r *http.Request) {
 	routeutils.WriteDataJson(w, string(quests))
 }
 
+func GetDailyQuestStatusProgress(w http.ResponseWriter, r *http.Request) {
+	userAddress := r.URL.Query().Get("address")
+	if userAddress == "" {
+		routeutils.WriteErrorJson(w, http.StatusBadRequest, "Missing address parameter")
+		return
+	}
+
+	dayIndexStr := r.URL.Query().Get("dayIndex")
+	if dayIndexStr == "" {
+		routeutils.WriteErrorJson(w, http.StatusBadRequest, "Missing dayIndex parameter")
+		return
+	}
+
+	dayIndex, err := strconv.Atoi(dayIndexStr)
+	if err != nil {
+		routeutils.WriteErrorJson(w, http.StatusBadRequest, "Invalid dayIndex parameter")
+		return
+	}
+
+	questJsons, err := core.PostgresQueryJson[QuestProgress](
+		`SELECT quest_id AS QuestId, quest_type as QuestType
+         FROM DailyQuests
+         WHERE day_index = $1`,
+		dayIndex,
+	)
+
+	if err != nil {
+		routeutils.WriteErrorJson(w, http.StatusInternalServerError, "Failed to get daily quest status")
+		return
+	}
+
+	var quests []QuestProgress
+	err = json.Unmarshal([]byte(questJsons), &quests)
```

Comment:
This like can be removed since you are calculating the result later

Reply:
removing this line will throw this 

`quest.QuestId undefined (type byte has no field or method QuestId)`

when looping over the data 

```
	var result []QuestProgress
	for _, quest := range questJsons {
		result = append(result, QuestProgress{
			QuestId:  quest.QuestId,
			Progress: 60,
			Needed:   40,
		})
	}

```

Reply:
Okay, I see thank you. I think you want to do `PostgresQuery` not `PostgresQueryJson`

Reply:
i will try it

Reply:
`PostgresQuery` clear the error of above